src/nyc_trees/apps/event/views.py

```diff
@@ -429,3 +445,26 @@ def increase_rsvp_limit(request, event_slug):
     return {
         'max_attendees': event.max_attendees
     }
+
+
+@transaction.atomic
+def event_email_unsubscribe(request, event_slug, token):
+    event = get_object_or_404(Event, group=request.group, slug=event_slug)
+
+    try:
+        username, signed_value = token.split(':', 1)
+    except ValueError:
+        # Invalid token
+        return {}
+
+    user = get_object_or_404(User, username=username)
+
+    if not user.is_valid_token(token):
+        # Invalid token
+        return {}
+
+    EventRegistration.objects.filter(event=event, user=user) \
+                             .update(opt_in_emails=False)
+
+    # Redirect to Emails tab on user profile page.
+    return redirect(reverse('user_profile_settings') + '#emails-pane')
```

Comment:
Decided to fix this by displaying a static page after unsubscribing to an event instead of redirecting to the user settings page.